SConstruct

```diff
@@ -748,3 +752,12 @@ if "env" in locals():
     # TODO: replace this with `env.Dump(format="json")`
     # once we start requiring SCons 4.0 as min version.
     methods.dump(env)
+
+
+def print_elapsed_time():
+    elapsed_time_sec = round(time.time() - time_at_start, 3)
+    time_ms = round((elapsed_time_sec % 1) * 1000)
+    print(f"[Time elapsed: {time.strftime('%H:%M:%S', time.gmtime(elapsed_time_sec))}.{time_ms:03}]")
```

Comment:
```suggestion
    elapsed_time_sec = round(time.time() - time_at_start * 1000) / 1000
    print(f"{str(int(elapsed_time_sec) // 3600).rjust(2, '0')}:{str(int(elapsed_time_sec) // 3600 // 60).rjust(2, '0')}:{str(int(elapsed_time_sec % 60 // 1)).rjust(2, '0')}{str(elapsed_time_sec % 60 - elapsed_time_sec % 60 // 1)[1:5].ljust(4, '0')}")
```
Solves the discussed problems

Reply:
This is unreadable nd overkill...

Reply:
@jiqz I think I already fixed it

Reply:
This does account for that case, and is slightly faster too.